pkg/workers/leader_election_mgr_test.go

```diff
@@ -185,3 +187,407 @@ func TestLeaderElectionManager_acquireLeaderLease(t *testing.T) {
+				},
+			},
+			want: &LeaderElectionManager{
+				workers:                                []Worker{},
+				connectionFactory:                      &db.ConnectionFactory{},
+				tearDown:                               nil,
+				leaderElectionReconcilerRepeatInterval: 15 * time.Second,
+				leaderLeaseExpirationTime:              1 * time.Minute,
+				workerGrp:                              sync.WaitGroup{},
+			},
+		},
+	}
+	for _, testcase := range tests {
+		tt := testcase
+		t.Run(tt.name, func(t *testing.T) {
+			g := NewWithT(t)
+			t.Parallel()
+			g.Expect(NewLeaderElectionManager(tt.args.workers, tt.args.connectionFactory, tt.args.reconcilerConfig)).To(Equal(tt.want))
+		})
+	}
+}
+
+func TestLeaderElectionManager_start(t *testing.T) {
+	type fields struct {
+		workers                                []Worker
+		connectionFactory                      *db.ConnectionFactory
+		tearDown                               chan struct{}
+		leaderElectionReconcilerRepeatInterval time.Duration
+		workerGrp                              *sync.WaitGroup
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		setupFn func()
+	}{
+		{
+			name: "should start the worker if the worker is a leader and is not running",
+			fields: fields{
+				workers: []Worker{
+					&WorkerMock{
+						GetIDFunc: func() string {
+							return "01"
+						},
+						GetWorkerTypeFunc: func() string {
+							return "cluster"
+						},
+						IsRunningFunc: func() bool {
+							return false
+						},
+						GetSyncGroupFunc: func() *sync.WaitGroup {
+							return &sync.WaitGroup{}
+						},
+						StartFunc: func() {},
+					},
+				},
+				connectionFactory:                      db.NewMockConnectionFactory(nil),
+				leaderElectionReconcilerRepeatInterval: 15 * time.Second,
+				workerGrp:                              &sync.WaitGroup{},
+				tearDown:                               make(chan struct{}),
+			},
+			setupFn: func() {
+				mockEntry := map[string]interface{}{
+					"leader":  "01",
+					"expires": time.Now().Add(time.Hour),
+				}
+				mocket.Catcher.Reset().
+					NewMock().
+					WithQuery("SELECT * FROM leader_leases").
+					WithReply([]map[string]interface{}{mockEntry})
+				mocket.Catcher.NewMock().WithExecException().WithQueryException()
+			},
+		},
+	}
+	for _, testcase := range tests {
+		tt := testcase
+		t.Run(tt.name, func(t *testing.T) {
+			g := NewWithT(t)
+			t.Parallel()
+			tt.setupFn()
+			s := &LeaderElectionManager{
+				workers:                                tt.fields.workers,
+				connectionFactory:                      tt.fields.connectionFactory,
+				tearDown:                               tt.fields.tearDown,
+				leaderElectionReconcilerRepeatInterval: tt.fields.leaderElectionReconcilerRepeatInterval,
+				workerGrp:                              sync.WaitGroup{},
+			}
+			s.Start()
+			g.Expect(tt.fields.workers[0].GetID()).To(Equal("01"))
+			g.Expect(tt.fields.workers[0].GetWorkerType()).To(Equal("cluster"))
```

Comment:
Here you are just calling the mock function you just created. I don't think this is what you wanted to do.
I think would be better to add something like: 
```suggestion
			g.Expect(tt.fields.workers[0].calls.GetID).To(HaveLen(1))
			g.Expect(tt.fields.workers[0].calls.GetWorkerType).To(HaveLen(1))
```